core/src/com/unciv/UncivGame.kt

```diff
@@ -172,6 +175,106 @@ class UncivGame(parameters: UncivGameParameters) : Game() {
         }
     }
 
+    fun nextTurn() {
+
+        Concurrency.runOnNonDaemonThreadPool("NextTurn") {
+
+            isTurnProcessing = true
+
+            debug("Next turn starting")
+            val startTime = System.currentTimeMillis()
+            val originalGameInfo = gameInfo!!
+            val gameInfoClone = originalGameInfo.clone()
+            gameInfoClone.setTransients()  // this can get expensive on large games, not the clone itself
+            gameInfoClone.nextTurn()
+
+            if (originalGameInfo.gameParameters.isOnlineMultiplayer) {
+                try {
+                    onlineMultiplayer.updateGame(gameInfoClone)
+                } catch (ex: Exception) {
+                    val message = when (ex) {
+                        is FileStorageRateLimitReached -> "Server limit reached! Please wait for [${ex.limitRemainingSeconds}] seconds"
+                        else -> "Could not upload game!"
+                    }
+                    launchOnGLThread { // Since we're changing the UI, that should be done on the main thread
+                        val cantUploadNewGamePopup = Popup(worldScreen!!)
+                        cantUploadNewGamePopup.addGoodSizedLabel(message).row()
+                        cantUploadNewGamePopup.addCloseButton()
+                        cantUploadNewGamePopup.open()
+                    }
+                    isTurnProcessing = false
+                    worldScreen?.shouldUpdate = true
+                    return@runOnNonDaemonThreadPool
+                }
+            }
+
+            isTurnProcessing = false
+
+            if (gameInfo != originalGameInfo)
+                return@runOnNonDaemonThreadPool  // while this was turning we loaded another game
+
+            debug("Next turn took %sms", System.currentTimeMillis() - startTime)
+            reloadWorldScreen(gameInfoClone)
+        }
+    }
```

Comment:
This shouldn't be here either, it's a game-specific function that's only relevant for the worldscreen

Reply:
I don't agree This function has nothing to do with worldscreen. UncivGame is the class who is basically responsible for main gamedata object (gameInfo) - current + cloned. It is his responsibility to do gameinfo copying, processing and managing worldscreen. WorldScreen itself is just responsible for rendering given data to UI and handling input from user.

We shouldn't conveniently put functions inside of object with other responsibilities

Reply:
Screens *are* personally responsible for doing work to bring you to other screens. That is part of their responsibility. UncivGame exists to initialize the first game state, and for convenience functions that affect multiple screens, neither of which this is